simulation/sub8_gazebo/diagnostics/gazebo_tests/align_marker_test.py

```diff
@@ -17,69 +18,89 @@ class Job(common.Job):
-        self.reset_grid = yield self.nh.get_service_client('/reset_occ_grid', Empty)
+    def initial_setup(self):
+        '''
+        Remove all the models in the sim and spawn a ground plane to run tests with.
+        TODO: Sometimes all the models don't get deleted correctly - probably a gazeb problem.
+        '''
+        models = yield self.model_states.get_next_message()
+
+        if 'ground_plane' not in models.name:
+            s = SpawnModelRequest()
+            s.model_name = 'ground_plane'
+            s.model_xml = ground_plane
+            s.initial_pose = msg_helpers.numpy_quat_pair_to_pose([0, 0, -5], [0, 0, 0, 1])
+            yield self.spawn_model(s)
+
+        for model in models.name:
+            if model == "channel_marker_1" or model == "sub8" or model == 'ground_plane':
+                continue
+            print "MARKER_TEST - Deleting {}".format(model)
+            self.delete_model(DeleteModelRequest(model_name=model))
+
+        self.nh.sleep(1)
 
+    @txros.util.cancellableInlineCallbacks
+    def setup(self):
         # Set sub position
-        sub_point = np.random.uniform(-10, 10, size=3)
+        sub_point = np.random.uniform(-5, 5, size=3)
         sub_point[2] = -np.abs(sub_point[2]) / 2.1
         sub_quat = tf.transformations.quaternion_from_euler(0, 0, np.random.uniform(-7, 7, size=1))
         sub_pose = msg_helpers.numpy_quat_pair_to_pose(sub_point, sub_quat)
         yield self.set_model_pose(sub_pose)
-        print "Sub at {0}".format(sub_pose)
+        print "MARKER_TEST - Sub at {0}".format(sub_pose)
 
-        # Reset grid
-        yield self.nh.sleep(.2)
-        yield self.reset_grid(EmptyRequest())
-        yield self.nh.sleep(1)
+        # Set marker position
+        self.marker_point = np.random.uniform(-3, 3, size=2)
+        self.marker_quat = tf.transformations.quaternion_from_euler(0, 0, np.random.uniform(-7, 7, size=1))
+        self.marker_pose = msg_helpers.numpy_quat_pair_to_pose(np.append(self.marker_point, -4.9), self.marker_quat)
+        print "MARKER_TEST - Marker at {0}".format(self.marker_pose)
+        self.set_model_pose(self.marker_pose, model='channel_marker_1')
 
     @txros.util.cancellableInlineCallbacks
     def run(self, sub):
         print "Running Mission"
-        # Set marker position
-        marker_point = np.random.uniform(-8, 8, size=2)
-        marker_quat = tf.transformations.quaternion_from_euler(0, 0, np.random.uniform(-7, 7, size=1))
-        marker_pose = msg_helpers.numpy_quat_pair_to_pose(np.append(marker_point, -4.9), marker_quat)
-        print "Marker at {0}".format(marker_pose)
-        self.set_model_pose(marker_pose, model='channel_marker_1')
 
         # Call actual mission
         start_time = self.nh.get_time()
         try:
-            resp = yield txros.util.wrap_timeout(missions.align_channel.run(sub), 600)
+            resp = yield txros.util.wrap_timeout(missions.align_channel.run(sub), 600)  # 10 minute timeout
```

Comment:
I suspect this is causing alot of my issues with weird errors that don't break anything (since it has in the past)